patches/api/0220-Add-methods-to-get-translation-keys.patch

```diff
@@ -393,6 +393,198 @@ index c799ac46dbe257d470d3b236cf55b41240f5fda8..d61855b83da0776fe910dee8cde184c7
+      */
+-    LECTERN(1, "Lectern"),
++    LECTERN(1, "Lectern", "container.lectern"), // Paper
+     /**
+      * A smoker inventory, with a RESULT slot, a CRAFTING slot, and a FUEL slot.
+      */
+-    SMOKER(3, "Smoker"),
++    SMOKER(3, "Smoker", "container.smoker"), // Paper
+     /**
+      * Loom inventory, with 3 CRAFTING slots, and 1 RESULT slot.
+      */
+-    LOOM(4, "Loom"),
++    LOOM(4, "Loom", "container.loom"), // Paper
+     /**
+      * Cartography inventory with 2 CRAFTING slots, and 1 RESULT slot.
+      */
+-    CARTOGRAPHY(3, "Cartography Table"),
++    CARTOGRAPHY(3, "Cartography Table", "container.cartography_table"), // Paper
+     /**
+      * Grindstone inventory with 2 CRAFTING slots, and 1 RESULT slot.
+      */
+-    GRINDSTONE(3, "Repair & Disenchant"),
++    GRINDSTONE(3, "Repair & Disenchant", "container.grindstone_title"), // Paper
+     /**
+      * Stonecutter inventory with 1 CRAFTING slot, and 1 RESULT slot.
+      */
+-    STONECUTTER(2, "Stonecutter"),
++    STONECUTTER(2, "Stonecutter", "container.stonecutter"), // Paper
+     /**
+      * Pseudo composter inventory with 0 or 1 slots of undefined type.
+      */
+-    COMPOSTER(1, "Composter")
++    COMPOSTER(1, "Composter", null) // Paper
+     ;
+ 
+     private final int size;
+@@ -142,6 +142,17 @@ public enum InventoryType {
+ 
+     // Paper start
+     private final net.kyori.adventure.text.Component defaultTitleComponent;
++    private final String translationKey;
++
++    /**
++     * @throws IllegalStateException if you call this on {@link #MERCHANT} or {@link #COMPOSTER}.
++     */
++    @Override
++    public @org.jetbrains.annotations.NotNull String translationKey() {
++        com.google.common.base.Preconditions.checkState(this.translationKey != null, this + " is not translatable");
++        return this.translationKey;
++    }
++
+ 
+     /**
+      * Gets the inventory's default title.
+@@ -151,16 +162,17 @@ public enum InventoryType {
+     public @NotNull net.kyori.adventure.text.Component defaultTitle() {
+         return defaultTitleComponent;
+     }
++    private InventoryType(int defaultSize, /*@NotNull*/ String defaultTitle, @org.jetbrains.annotations.Nullable String translationKey) {
++        this(defaultSize, defaultTitle, true, translationKey);
+     // Paper end
+-    private InventoryType(int defaultSize, /*@NotNull*/ String defaultTitle) {
+-        this(defaultSize, defaultTitle, true);
+     }
+ 
+-    private InventoryType(int defaultSize, /*@NotNull*/ String defaultTitle, boolean isCreatable) {
++    private InventoryType(int defaultSize, /*@NotNull*/ String defaultTitle, boolean isCreatable, @org.jetbrains.annotations.Nullable String translationKey) { // Paper
+         size = defaultSize;
+         title = defaultTitle;
+         this.isCreatable = isCreatable;
+         this.defaultTitleComponent = net.kyori.adventure.text.Component.text(defaultTitle); // Paper - Adventure
```

Comment:
Maybe this should be updated, but ideally the title would be nullable if you don't want to break the api you could fallback to the hardcoded value for inventory that doesn't have a translation key

Reply:
idk if it would even be breaking the API. it returns a component, no guarantees of what type of component.